src/main/java/lotto/domain/Rank.java

```diff
@@ -0,0 +1,48 @@
+package lotto.domain;
+
+import java.util.Arrays;
+
+public enum Rank {
+    UNRANKED(0L, 0, false),
+    FIFTH(5000L, 3, false),
+    FOURTH(50000L, 4, false),
+    THIRD(1500000L, 5, false),
+    SECOND(30000000L, 5, true),
+    FIRST(2000000000L, 6, false);
+
+    private Long price;
+    private Integer criterion;
+    private boolean requireBonus;
+
+    Rank(Long price, Integer criterion, boolean requireBonus) {
+        this.price = price;
+        this.criterion = criterion;
+        this.requireBonus = requireBonus;
+    }
```

Comment:
(이런 방법도 있어요!)

requireBonus같은 경우 2등을 제외하면 나머지는 전부 false입니다. 그러면 요소가 늘어날 때 마다 false를 추가해야 하는데 다음과 같이 작성하는 방법도 고려해볼 수 있어요!

```suggestion
    UNRANKED(0L, 0),
    FIFTH(5000L, 3),
    FOURTH(50000L, 4),
    THIRD(1500000L, 5),
    SECOND(30000000L, 5, true),
    FIRST(2000000000L, 6);

    private Long price;
    private Integer criterion;
    private boolean requireBonus;

    Rank(Long price, Integer criterion, boolean requireBonus) {
        this.price = price;
        this.criterion = criterion;
        this.requireBonus = requireBonus;
    }
    Rank(Long price, Integer criterion) {
        this.price = price;
        this.criterion = criterion;
        this.requireBonus = false;
    }
```

Reply:
> (이런 방법도 있어요!)

rank에서 보너스만 어떻게 특정 지을지 고민을 하다가 결국엔 객체의 필드로 만들었는데,
enum의 생성자를 통해 보너스가 불필요한 당첨 금액을 중복코드 없이 구분할 수 있군요!
좋은 정보 감사합니다